packages/webcomponents/src/components/business-form-stepped/legal-address-form/legal-address-form-step.tsx

```diff
@@ -11,101 +16,164 @@ import { FormController } from '../../form/form';
   styleUrl: 'legal-address-form-step.scss',
 })
 export class LegalAddressFormStep {
-  @Prop() formController: FormController;
+  @Prop() authToken: string;
+  @Prop() businessId: string;
+  @State() formController: FormController;
   @State() errors: any = {};
   @State() legal_address: any = {};
+  @Event({ bubbles: true }) submitted: EventEmitter<{ data?: any }>;
+  @Event() formLoading: EventEmitter<boolean>;
+  @Event() serverError: EventEmitter<{ data?: any, message?: string }>;
 
   constructor() {
     this.inputHandler = this.inputHandler.bind(this);
+    this.sendData = this.sendData.bind(this);
+    this.fetchData = this.fetchData.bind(this);
+  }
+
+  private api: any;
+
+  get businessEndpoint() {
+    return `entities/business/${this.businessId}`
+  }
+
+  private async fetchData() {
+    this.formLoading.emit(true);
+    try {
+      const response: IApiResponse<IBusiness> = await this.api.get(this.businessEndpoint);
+      this.legal_address = response.data.legal_address;
```

Comment:
I think we could use the Address class here, this way we'll always have a country set. 
I tried to edit a business created from Insomnia, and got stuck on the second step because the business didn't have a country.

Reply:
Good call - thanks man!